### PR TITLE
add env to control torch-blade cluster iteration

### PR DESCRIPTION
**Pull request by linbinskn**
Current torch_blade cluster setting(`max_iter_count=10`) has risk of early stopping before cluster final convergence. In the case of bert-base with fakequant node, the number of fusion block is 62 when `max_iter_count=10` while the number is 7 when `max_iter_count=40`. This phenomenon drastically influence the latency.
This PR aims to prevent performance drop in some cases by adding env to set specific `max_iter_count` value. Clustering policy refinement is in TODO list.

### CODE REVIEW

File: pytorch_blade/torch_blade/clustering/support_fusion_algorithm.py [hunk 1]
```diff
@@ -189,7 +190,9 @@ def can_merge(u, v):
         return True
 
     graph_topolist = graph_builder.group_topolist()
-    max_iter_count = 10
+    # some graph unions may not converge in 10 iterations, provide customize setting from env
+    # TODO: refine cluster policy
+    max_iter_count = int(os.getenv('TORCH_CLUSTER_MAX_ITER_COUNT', 10))
```
Comment (qiuxiafei) [inline pytorch_blade/torch_blade/clustering/support_fusion_algorithm.py:195]: I think a config option is torch_blade's convention.

**Comment by qiuxiafei** [comment]
What's the initial idea of fixed iteration number instead of simply run to convergence? @tanyokwok

File: pytorch_blade/torch_blade/clustering/support_fusion_algorithm.py (hunk 1 above)
Comment (linbinskn) [reply pytorch_blade/torch_blade/clustering/support_fusion_algorithm.py]: Good point, done.

**Review by qiuxiafei — APPROVED** [review]

**Comment by tanyokwok** [comment]
> What's the initial idea of fixed iteration number instead of simply run to convergence?

It's set to avoid too long and dead loops. Anyway, I think it's time to refine the subgraph capture algorithms in TorchBlade.

**Pull request merged**